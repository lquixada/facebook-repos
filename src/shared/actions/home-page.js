export const HOME_PAGE_REQUESTED = 'HOME_PAGE_REQUESTED'
export const HOME_PAGE_SUCCEEDED = 'HOME_PAGE_SUCCEEDED'
export const HOME_PAGE_FAILED = 'HOME_PAGE_FAILED'

export const loadHomePage = () => ({
  type: HOME_PAGE_REQUESTED
})
