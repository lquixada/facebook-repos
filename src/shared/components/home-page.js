import React from 'react';
import styled from 'styled-components';

import Menu from '../containers/menu';
import {Footer} from './footer';
import {Title} from './title';
import {Main, Aside, Content, Header, HeaderContainer, Section, SectionContainer} from './grid';
import {flex} from '../helpers';

export const HomePage = () => (
  <Main>
    <Header>
      <HeaderContainer>
        <Title />
      </HeaderContainer>
    </Header>

    <Section>
      <SectionContainer>
        <Aside>
          <Menu />
          <Footer />
        </Aside>
        <Content>
          <Wrapper>
            Choose a repository on the menu.
          </Wrapper>
        </Content>
      </SectionContainer>
    </Section>
  </Main>
);

const Wrapper = styled.div`
  ${flex.display()}
  ${flex.middle()} 
  ${flex.center()}
  height: 50rem;
  border: .7rem dashed #eee;
  border-radius: .6rem;
  font-size: 1.4rem;
  letter-spacing: -.1px;
`;
