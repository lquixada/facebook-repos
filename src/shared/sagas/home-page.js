import {take, fork} from 'redux-saga/effects'

import {HOME_PAGE_REQUESTED} from '../actions'
import {loadReposContributorsCount} from './repos-contributors-count'

export default function * watchLoadHomePage () {
  while (true) {
    yield take(HOME_PAGE_REQUESTED)

    yield fork(loadReposContributorsCount)
  }
}
