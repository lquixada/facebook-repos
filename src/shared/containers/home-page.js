import React from 'react'
import {compose} from 'redux'
import {connect} from 'react-redux'
import {HomePage} from '../components/home-page'
import {provideHooks} from '../helpers'
import * as actions from '../actions'

const hooks = {
  fetch: ({params, dispatch}) => dispatch(actions.fetchPage({
    name: 'home',
    owner: params.owner || 'facebook'
  }))
}

export class HomePageContainer extends React.Component {
  componentDidMount () {
    this.fetch()
  }

  componentDidUpdate (prevProps) {
    if (this.hasChanged(prevProps)) {
      this.fetch()
    }
  }

  fetch () {
    this.props.fetchPage({
      name: 'home',
      owner: this.props.owner || 'facebook'
    })
  }

  hasChanged (prevProps) {
    return (
      prevProps.owner !== this.props.owner
    )
  }

  render () {
    return (
      <HomePage />
    )
  }
}

const mapStateToProps = (state, {match}) => ({
  owner: match.params.owner,
  repoName: match.params.repo
})
const composed = compose(provideHooks(hooks), connect(mapStateToProps, actions))
export default composed(HomePageContainer)
