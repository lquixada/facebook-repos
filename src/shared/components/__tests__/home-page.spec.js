import React from 'react';
import {shallow} from 'enzyme';

import {HomePage} from '../home-page';

describe('<HomePage />', () => {
  it('matches the snapshot', () => {
    const component = shallow(<HomePage />);
    expect(component).toMatchSnapshot();
  });
});
