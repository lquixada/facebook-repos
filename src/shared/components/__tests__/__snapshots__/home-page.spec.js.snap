// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> matches the snapshot 1`] = `
<styled.main>
  <styled.header>
    <styled.div>
      <Title />
    </styled.div>
  </styled.header>
  <styled.section>
    <styled.div>
      <styled.aside>
        <withRouter(Connect(Menu)) />
        <Footer />
      </styled.aside>
      <styled.article>
        <styled.div>
          Choose a repository on the menu.
        </styled.div>
      </styled.article>
    </styled.div>
  </styled.section>
</styled.main>
`;
